### PR TITLE
Allow OpINotEqual in OpSpecConstantOp inst

### PR DESCRIPTION
**Pull request by Qining**
Pendding for now until the spec is fixed.

### CODE REVIEW

File: source/assembly_grammar.cpp [hunk 1]
```diff
@@ -165,8 +166,8 @@ const SpecConstantOpcodeEntry kOpSpecConstantOpcodes[] = {
     CASE(InBoundsPtrAccessChain),
 };
 
-// The 58 is determined by counting the opcodes listed in the spec.
-static_assert(58 == sizeof(kOpSpecConstantOpcodes)/sizeof(kOpSpecConstantOpcodes[0]),
```
Comment (antiagainst) [inline source/assembly_grammar.cpp]: Can use `ARRAY_SIZE` in `macro.h`

**Comment by dneto0** [comment]
Let's wait on submitting this until we either get feedback from the SPIR working group, or we hear about failures in the field.

**Comment by Qining** [comment]
Rebased. Ready for submit.

**Comment by dneto0** [comment]
+2

**Pull request merged**